tools/src/main/java/org/nmdp/ngs/tools/ValidateInterpretation.java

```diff
@@ -47,105 +49,118 @@
-			                            result = true;
-			                        }
-			                    }
-			                    
-	
-			                    if (result) {
-			                        passes++;
-			                    }
-			                    else {
-			                        failures++;
-			                    }
-			
-			                    if (!printSummary) {
-			                        writer.println((result ? "PASS" : "FAIL") + "\t" + sample + "\t" + expectedAllele + "\t" + expected.get(sample).getExperiment());
-			                    }
-			                }
-	            		}
-            		}
-            	}
-            }
 
+            // interpretations keyed by sample
+            ListMultimap<String, Interpretation> expected = read(expectedFile);
+            ListMultimap<String, Interpretation> observed = read(observedFile);
+
+            // for each sample in observed
+            for (String sample : observed.keySet()) {
+                for (Interpretation o : observed.get(sample)) {
+                    AlleleList observedAlleleList = asAlleleList(o);
+                    if (shouldValidate(o, observedAlleleList)) {
+
+                        // for each matching sample in expected
+                        for (Interpretation e : expected.get(sample)) {
+                            Genotype expectedGenotype = asGenotype(e);
+                            for (Haplotype expectedHaplotype : expectedGenotype.getHaplotypes()) {
+                                for (AlleleList expectedAlleleList : expectedHaplotype.getAlleleLists()) {
+                                    if (sameLocus(observedAlleleList, expectedAlleleList)) {
+
+                                        // for each pair of expected and observed alleles
+                                        for (Allele expectedAllele : expectedAlleleList.getAlleles()) {
+                                            boolean match = false;
+
+                                            for (Allele observedAllele : observedAlleleList.getAlleles()) {
+                                                if (matchByField(expectedAllele.getGlstring(), observedAllele.getGlstring()) >= resolution) {
+                                                    match = true;
+                                                    break;
+                                                }
+                                            }
+
+                                            if (match) {
+                                                passes++;
+                                            }
+                                            else {
+                                                failures++;
+                                            }
+
+                                            if (!printSummary) {
+                                                writer.println((match ? "PASS" : "FAIL") + "\t" + sample + "\t" + expectedAllele);
```

Comment:
Do we want to add the experiment to this? I was thinking it may be easier to recombine things in the nextflow script if it's in here, but I haven't gotten very far with that yet so I'm sure.